### PR TITLE
Remove resolveHandle and IFluidHandleContext from ContainerRuntime interfaces

### PR DESCRIPTION
**Pull request by kian-thompson**
### Removed `resolveHandle` and `IFluidHandleContext` from ContainerRuntime interfaces

The `IContainerRuntime.resolveHandle(...)` method and the `IContainerRuntimeBase.IFluidHandleContext` property have been removed. Please remove all usage of these APIs.

See [Removing-IFluidRouter.md](https://github.com/microsoft/FluidFramework/blob/main/packages/common/core-interfaces/Removing-IFluidRouter.md) for more details.

[AB#5593](https://dev.azure.com/fluidframework/internal/_workitems/edit/5593)

### CODE REVIEW

File: packages/test/test-service-load/src/loadTestDataStore.ts [hunk 1]
```diff
@@ -749,7 +749,8 @@ const LoadTestDataStoreInstantiationFactory = new DataObjectFactory(
 );
 
 const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
-	runtime.IFluidHandleContext.resolveHandle(request);
+	// This cast is safe as ContainerRuntime.loadRuntime is used in ContainerRuntimeFactoryWithDefaultDataStore
+	(runtime as ContainerRuntime).resolveHandle(request);
```
Comment (agarwal-navin) [inline packages/test/test-service-load/src/loadTestDataStore.ts:753]: Why is this needed? The service load tests don't run in compat mode right?

File: packages/test/test-end-to-end-tests/src/test/dataStoreRetrieval.spec.ts [hunk 2]
```diff
@@ -87,7 +88,7 @@ describeFullCompat(
 
 		let provider: ITestObjectProvider;
 		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
-			runtime.IFluidHandleContext.resolveHandle(request);
+			(runtime as ContainerRuntime).resolveHandle(request);
```
[the remark this answers is not in the document]
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/dataStoreRetrieval.spec.ts]: I initially left them for back-compat, but I should've fixed all the weird cases when I went through all the tests the other week

**Review by agarwal-navin — APPROVED** [review]
